err.go

```diff
@@ -21,15 +21,20 @@ import "fmt"
 type InvalidOperationError struct {
 	structName string
 	methodName string
+	message    string
 }
 
 func (e *InvalidOperationError) Error() string {
-	return fmt.Sprintf(`ionhash: invalid operation error on %v.%v`, e.structName, e.methodName)
+	if e.message != "" {
+		return fmt.Sprintf(`ionhash: %v.%v: %v`, e.structName, e.methodName, e.message)
```

Comment:
For the generic case I would add something 
```
`ionhash: Invalid operation at %v,%v: %v`
```

To distinguish if from just printing information.

Reply:
Done.